packages/flutter/lib/src/widgets/framework.dart

```diff
@@ -3287,29 +3287,27 @@ abstract class Element extends DiagnosticableTree implements BuildContext {
   @mustCallSuper
   void didChangeDependencies() {
     assert(_active); // otherwise markNeedsBuild is a no-op
-    _debugCheckOwnerBuildTargetExists('didChangeDependencies');
+    assert(() {
+      _debugCheckOwnerBuildTargetExists('didChangeDependencies');
+      return true;
+    });
     markNeedsBuild();
   }
 
   void _debugCheckOwnerBuildTargetExists(String methodName) {
-    assert(() {
-      if (owner._debugCurrentBuildTarget == null) {
-        throw new FlutterError(
-          '$methodName for ${widget.runtimeType} was called at an '
-          'inappropriate time.\n'
-          '\n'
-          'It may only be called while the widgets are being built. A possible '
-          'cause of this error is when $methodName is called during '
-          'one of:\n'
-          '\n'
-          ' * network I/O event\n'
-          ' * file I/O event\n'
-          ' * timer\n'
-          ' * microtask (caused by Future.then, async/await, scheduleMicrotask)'
-        );
-      }
-      return true;
-    });
+    if (owner._debugCurrentBuildTarget == null) {
```

Comment:
this should still be in an assert, so that it can't be called in release builds

Reply:
Done. Just so I understand, we want both the call to `_debugCheckOwnerBuildTargetExists` be wrapped in an `assert` _and_ wrap the contents of `_debugCheckOwnerBuildTargetExists` in _another_ `assert`, correct?

Reply:
Right. In release builds there should be no code to compile.